ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -1223,14 +1224,12 @@ func isServeAppUnhealthyOrDeployedFailed(appStatus string) bool {
 
 // TODO: Move this function to util.go and always use this function to retrieve the head Pod.
 func (r *RayServiceReconciler) getHeadPod(ctx context.Context, instance *rayv1.RayCluster) (*corev1.Pod, error) {
-	podList := corev1.PodList{}
-	if err := r.List(ctx, &podList, common.RayClusterHeadPodsAssociationOptions(instance).ToListOptions()...); err != nil {
-		return nil, err
+	if instance.Status.Head.PodName == "" {
+		return nil, fmt.Errorf("Found 0 head pods for RayCluster %s in the namespace %s", instance.Name, instance.Namespace)
```

Comment:
This error can probably be updated "RayCluster status did not contain head pod name" since we're not listing pods anymore

Reply:
How about "no PodName in the .Status.Head for RayCluster %s in the namespace %s"?

Reply:
Or "RayCluster %s in the namespace %s did not contain .Status.Head.PodName"

Reply:
both sound good to me!